components/org.wso2.carbon.identity.oauth.scope.endpoint/src/test/java/org/wso2/carbon/identity/oauth/scope/endpoint/util/ScopeUtilsTest.java

```diff
@@ -0,0 +1,201 @@
+
+    @Test(description = "Testing getScopeDTO")
+    public void testGetScopeDTO() throws Exception {
+        Scope scope = mock(Scope.class);
+        when(scope.getBindings()).thenReturn(new ArrayList<String>());
+        when(scope.getDescription()).thenReturn(Description);
+        when(scope.getName()).thenReturn(name);
+
+        ScopeDTO scopeDTOexpected = new ScopeDTO();
+        scopeDTOexpected.setBindings(new ArrayList<String>());
+        scopeDTOexpected.setDescription(Description);
+        scopeDTOexpected.setName(name);
+
+        ScopeDTO scopeDTO1 = ScopeUtils.getScopeDTO(scope);
+        Assert.assertEquals(scopeDTO1.getBindings(), scopeDTOexpected.getBindings(), errorBinding);
+        Assert.assertEquals(scopeDTO1.getDescription(), scopeDTOexpected.getDescription(), errorDescription);
+        Assert.assertEquals(scopeDTO1.getName(), scopeDTOexpected.getName(), errorName);
+
+    }
+
+    @Test(description = "Testing getUpdateScope")
+    public void testGetUpdatedScope() throws Exception {
+        ScopeToUpdateDTO scopeToUpdateDTO = mock(ScopeToUpdateDTO.class);
+        ArrayList<String> scope = new ArrayList<>(Arrays.asList("scope1", "scope2"));
+        when(scopeToUpdateDTO.getDescription()).thenReturn(Description);
+        when(scopeToUpdateDTO.getBindings()).thenReturn(scope);
+
+        Scope scopeexpected = new Scope(name, Description, scope);
+        scopeexpected.setDescription(Description);
+        scopeexpected.setBindings(scope);
+        ScopeToUpdateDTO sc = new ScopeToUpdateDTO();
+
+        Scope scope1 = ScopeUtils.getUpdatedScope(scopeToUpdateDTO, errorName);
+        Assert.assertEquals(scope1.getDescription(), scopeexpected.getDescription(), errorDescription);
+        Assert.assertEquals(scope1.getBindings(), scopeexpected.getBindings(), errorBinding);
+
+    }
+
+    @Test(description = "Testing getScopeDTO")
+    public void testGetScopeDTOs() throws Exception {
+        int scopeName;
+        int scopeSize = 15;
+        Set<Scope> scopes = new HashSet<>();
+        ArrayList<String> scope = new ArrayList<>(Arrays.asList("scope1", "scope2"));
+        for (scopeName = 0; scopeName < scopeSize; scopeName++) {
+            Scope scope1 = new Scope(name, Description, scope);
+            scopes.add(scope1);
+        }
+        Set<ScopeDTO> scopeDTOs = ScopeUtils.getScopeDTOs(scopes);
+        Assert.assertNotNull(scopeDTOs);
+        Assert.assertEquals(scopeDTOs.size(), scopeSize, "Scopes can't be found");
+
+    }
+
+    @DataProvider(name = "BuildScopeEndpointException")
+    public Object[][] buildScopeEndpointException() {
+        Response.Status status = Response.Status.BAD_REQUEST;
+        Log log = LogFactory.getLog(ScopeUtils.class);
+        Throwable throwable1 = new ScopeEndpointException(status);
+        Throwable throwable2 = new RuntimeException("BAD_REQUEST_INVALID_REDIRECT_URI");
+
+        return new Object[][]{
+                {status, "happy", throwable1, true, log},
```

Comment:
Let's remove string "happy" and log from data provider. Basically they don't change. Hence no need to give through data provider.